pandas/core/indexing.py

```diff
@@ -2222,6 +2224,8 @@ def check_bool_indexer(index: Index, key) -> np.ndarray:
                 "the indexed object do not match)."
             )
         result = result.astype(bool)._values
+    elif is_extension_array_dtype(key) and is_bool_dtype(key):
```

Comment:
why do we have this at all anymore? is check_array_indexer not sufficient ?

Reply:
Yeah good point, there's a lot of duplicate code happening. Will work on cleaning this up.